core/src/internal/rpc/mod.rs

```diff
@@ -3,25 +3,74 @@ use scaii_defs::protos::{MultiMessage, ScaiiPacket};
 use scaii_defs::protos::{ModuleInit, RpcConfig};
 use std::error::Error;
 use std::net::{IpAddr, SocketAddr};
-
 use websocket::client::sync::Client;
 use websocket::stream::sync::TcpStream;
 
 use scaii_defs::protos::endpoint::Endpoint;
-
+use std::process::Child;
+use std::process::Command;
 use super::LoadedAs;
+use scaii_defs::protos::init_as::InitAs;
+use scaii_defs::protos;
+use std::thread;
+use std::sync::mpsc;
 
 #[cfg(test)]
 mod test;
 
 pub fn init_rpc(rpc_config: RpcConfig) -> Result<LoadedAs, Box<Error>> {
-    use scaii_defs::protos::init_as::InitAs;
-    use scaii_defs::protos::endpoint::Endpoint;
-    use scaii_defs::protos;
+    let (tx, rx) = mpsc::channel();
+    let command = rpc_config.command.clone();
+    match command {
+        None => (),
+        Some(cmd) => {
+            let args = rpc_config.command_args.clone();
+            let _handle = thread::spawn(move || {
+                launch_far_end(&cmd, args.to_vec());
+                rx.recv().unwrap();
+            });
+        }
+    }
 
+    let result = startup_module(&rpc_config);
+    tx.send(String::from("rpc_started")).unwrap();
+    result
+}
+// pub fn init_rpc_broken(rpc_config: RpcConfig) -> Result<LoadedAs, Box<Error>> {
```

Comment:
Again, I'd prefer not to merge a big commented block of code. Can you move it to a new branch or delete it?

Reply:
will do